### PR TITLE
Pull requests on Gitlab

### LINKED ISSUE #414: Pull requests on Gitlab projects
By looking at the code I noticed there are no tests for creating pull requests in different scenarios (like Github has) and also the code doesn't seem to be compliant to the documentation of ogr.

-  [ ] Add tests
-  [ ] Fix the implementation

May be blocked by #412

### PR DESCRIPTION
**Pull request by mfocko**
Closes #414

### CODE REVIEW

File: ogr/services/gitlab/pull_request.py [hunk 1]
```diff
@@ -118,16 +119,50 @@ def create(
...
+            other_project = GitlabPullRequest.__get_fork(
+                fork_username,
+                project.get_forks()
+                if project.parent is None
+                else project.parent.get_forks(),
+            )
+
+            parameters["target_project_id"] = other_project.gitlab_repo.attributes["id"]
+            if project.parent is None:
+                parameters["target_project_id"] = repo.attributes["id"]
```
Comment (mfocko) [inline ogr/services/gitlab/pull_request.py:146]: maybe use one variable `target_id: Optional[int]` and just assign it if has value at the end? Would be cleaner

File: ogr/services/gitlab/pull_request.py [hunk 2]
```diff
@@ -118,16 +119,50 @@ def create(
...
+                repo = other_project.gitlab_repo
+
+        mr = repo.mergerequests.create(parameters)
         return GitlabPullRequest(mr, project)
 
+    @staticmethod
+    def __get_fork(
+        fork_username: str, forks: List["ogr_gitlab.GitlabProject"]
```
Comment (mfocko) [inline ogr/services/gitlab/pull_request.py:154]: just pass the project in to move `get_forks` call here?

File: ogr/services/gitlab/pull_request.py [hunk 3]
```diff
@@ -118,16 +119,50 @@ def create(
...
+            "description": body,
+        }
+
+        if project.is_fork and (
+            fork_username is None or fork_username == project.namespace
+        ):
+            parameters["target_project_id"] = project.parent.gitlab_repo.attributes[
+                "id"
+            ]
+        elif fork_username and fork_username != project.namespace:
```
Comment (lachmanfrantisek) [inline ogr/services/gitlab/pull_request.py:136]: Maybe some comment about `else` variant?

File: ogr/services/gitlab/pull_request.py (hunk 2 above)
Comment (lachmanfrantisek) [reply ogr/services/gitlab/pull_request.py:154]: Yes, sounds good. Or, document/name this method properly. It takes a while to understand what is going on here.

(e.g. Why you search for fork and you already send a list of forks... It's more about searching for the right fork than getting the fork(s).)

File: ogr/services/gitlab/pull_request.py [hunk 4]
```diff
@@ -118,16 +119,50 @@ def create(
...
+    ) -> "ogr_gitlab.GitlabProject":
+        forks = list(
+            filter(
+                lambda fork: fork.gitlab_repo.namespace["full_path"] == fork_username,
+                forks,
+            )
+        )
+        if not forks:
+            raise GitlabAPIException("Requested fork doesn't exist")
+        return forks[0]
```
Comment (lachmanfrantisek) [inline ogr/services/gitlab/pull_request.py:164]: Is there a chance that we get more than one?

File: tests/integration/test_gitlab.py [hunk 5]
```diff
@@ -552,6 +552,92 @@ def test_source_project_renamed_upstream(self):
         assert source_project.namespace == "mfocko"
         assert source_project.repo == "new-ogr-testing-repo-in-the-group"
 
+    def test_create_pr_upstream_upstream(self):
```
Comment (lachmanfrantisek) [inline tests/integration/test_gitlab.py:555]: Thanks for all those tests.

File: ogr/services/gitlab/pull_request.py (hunk 4 above)
Comment (mfocko) [reply ogr/services/gitlab/pull_request.py:164]: It filters by username, so it shouldn't.
However after GitLab allowing to create MR from non-existing branch through API, I wouldn't be surprised :D 

More straight-forward would be to get the project directly, but that introduces problem with renamed project (either one).

File: ogr/services/gitlab/pull_request.py (hunk 3 above)
Comment (mfocko) [reply ogr/services/gitlab/pull_request.py:136]: I'll add those, also usage as in github implementation

**Pull request merged**